src/mod/codecs/mod_opus/opus_parse.c

```diff
@@ -0,0 +1,355 @@
+			/* We should not go there as CELT_ONLY is already tested above */
+			return FALSE;
+		}
+
+		packet_info->frames_silk = silk_frame_nb_flags;
+		silk_size_frame_ms_per_flag = switch_opus_get_silk_frame_ms_per_flag(packet_info->config, silk_frame_nb_flags); /* 10 or 20 ms frame*/
+		if (!silk_size_frame_ms_per_flag) {
+			/* we should not go there as CELT_ONLY is already tested above */
+			return FALSE;
+		}
+
+		ptr_LBBR_FLAGS = packet_LBBR_FLAGS;
+		ptr_VAD_FLAGS = packet_VAD_FLAGS;
+
+		for (f = 0; f < packet_info->frames; f++) {
+			switch_opus_get_VAD_LBRR_flags(frame_data[f], silk_frame_nb_flags, ptr_VAD_FLAGS,  ptr_LBBR_FLAGS,
+				&silk_frame_nb_vad1, &silk_frame_nb_fec);
+			packet_info->vad += silk_frame_nb_vad1;
+			packet_info->fec += silk_frame_nb_fec;
+			packet_info->vad_ms += silk_frame_nb_vad1 * silk_size_frame_ms_per_flag;
+			packet_info->fec_ms += silk_frame_nb_fec * silk_size_frame_ms_per_flag;
+
+			ptr_VAD_FLAGS += silk_frame_nb_flags;
+			ptr_LBBR_FLAGS += silk_frame_nb_flags;
+		}
+		/* store the VAD & LBRR flags of all 20 ms silk-frames of the packet; LSB the first frame, MSB: the last */
+		vad_flags_per_silk_frame = 0;
+		fec_flags_per_silk_frame = 0;
+		silk_frame_packet = packet_info->frames * packet_info->frames_silk;
+		if (silk_frame_packet > 15) {
+			if (debug) {
+				switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_DEBUG, "opus_packet_parse: more than %d 20-ms frames in the packet ; only first 15 silk-frames data will be stored (pb silkFastAccelerate)\n", silk_frame_packet);
+			}
+			silk_frame_packet = 15;
+		}
+		ptr_LBBR_FLAGS = packet_LBBR_FLAGS;
+		ptr_VAD_FLAGS = packet_VAD_FLAGS;
+		shift_silk = 0;
+		for (i=0; i < silk_frame_packet; i++) {
+			vad_flags_per_silk_frame += (*ptr_VAD_FLAGS) << shift_silk;
+			fec_flags_per_silk_frame += (*ptr_LBBR_FLAGS) << shift_silk;
+			shift_silk++;
+			ptr_LBBR_FLAGS++; ptr_VAD_FLAGS++;
+		}
+		packet_info->vad_flags_per_silk_frame = vad_flags_per_silk_frame;
+		packet_info->fec_flags_per_silk_frame = fec_flags_per_silk_frame;
+		return TRUE;
+	}
+
+	if (packet_info->config != 1 && packet_info->config != 5 && packet_info->config != 9) {
+		if (debug) {
+			switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_DEBUG, "opus_packet_parse: the current parser implementation does not support muliple SILK frames for VAD or FEC detection.\n");
+		}
+		return FALSE;
+	}
+	/*
+	 *  Parse the VAD and LBRR flags in each Opus frame
+	 * */
+	for (f = 0; f < packet_info->frames; f++) {
+		if (frame_data[f][0] & 0x80) {
+			packet_info->vad++;
+		}
+		if (frame_data[f][0] & 0x40) {
+			packet_info->fec++;
+		}
+		if (debug) {
+			switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_DEBUG, "opus_packet_parse: LP layer opus_frame[%d] VAD[%d] FEC[%d]\n", f+1, (frame_data[f][0]&0x80)>>7, (frame_data[f][0]&0x40)>>6);
+		}
+	}
+	packet_info->vad_ms = packet_info->vad * packet_info->ms_per_frame;
+	packet_info->fec_ms = packet_info->fec * packet_info->ms_per_frame;
+	return TRUE;
+}
```

Comment:
please add the footer about editor settings:
https://developer.signalwire.com/freeswitch/FreeSWITCH-Explained/Community/Contributing-Code/Coding-Guidelines_6586862/